### ISSUE TITLE
Inappropriate values for `Concept_In_Subset` showing up in content

### ISSUE THREAD

**Issue by bdionne** [NCIEVS/nci-protege5#347]
````
For the each of the past few of months, we have had inappropriate Concept_In_Subset or Contributing_Source filler values creep into protégé. (This has occurred about once per month since September or October.) I think there is an issue where the annotation property editing dialog doesn’t sync the property with the correct value type and I believe this issue is related to the fact that the dialog remembers the last property and value the editor added. I have seen this sync issue before when I have been editing but I don’t know exactly how to replicate the issue.
````

**Comment by NCIEVS** [comment]
From [EHD] in last review.

EHD. Yes, this does still happen. This month, I managed to put Concept_In_Subset values into the Contributing_Source property. Still not sure how to replicate the problem (as this can occur due to inadvertent misuse of the Add Annotation dialog). I would like the dialog to be able to vet the values based on the chosen property so invalid values cannot be committed but I don't know what kind of effort that would take.

NT. this may be user error, not noting which annotation is active

**Comment by singhik** [comment]
@bdionne @NCIEVS : Hi, not sure on exactly what this issue was and  the fix applied ? 

I am still able to add Concept IRIs or rdfs Labels to 'Contributing_Source' as property (per attached ) ? 

![image](https://user-images.githubusercontent.com/36314371/129617826-9e8f7649-b9a2-42ad-99a0-447afc0ba3da.png)

**Comment by bdionne** [comment]
The issue here @singhik is that when you edit these properties, the tabs available should be governed by the type. So `Contributing_Source` is defined with a type of an enum. When you edit that you should only see and be able to use the enum tab and forced to select form the combo box

For `Concept_In_Subset` you should only be able to use the class hierarchy tab or the tab that allows you to enter an IRI explicit.

I can show you over a webex if this isn't clear

**Comment by NCIEVS** [comment]
@singhik , yes, that looks fine, the rdfs:comment is an annotation on the contributing_source annotation.  We don't disallow those.

**Comment by singhik** [comment]
@bdionne @NCIEVS   :Hi,  How about restricting (enabling of tabs) for 'Add annotations' as  option ? 'Add annotations' does allow all operations (tabs) for all properties without any restriction ?  One of the comments from Editors above was **"Still not sure how to replicate the problem (as this can occur due to inadvertent misuse of the Add Annotation dialog)"**  . So has 'Add annotations' (@) been handled as part of this implementation besides 'Edit ' ? 


![image](https://user-images.githubusercontent.com/36314371/129775580-d9404514-f867-490a-9ba8-648dbe65be27.png)

**Comment by bdionne** [comment]
@singhik That comment about misuse of Add Annotation is referring to the top level addition of annotations on a class, *not* the addition of annotations on annotations.

We could consider supporting it there as well, but that's a separate requirement and would involve changes to protégé
